### PR TITLE
C++ PointerEvent Demo (faster lasers on tablet/overlays)

### PR DESCRIPTION
**Pull request by SamGondelman**
Quick and dirty demo:
- RayPicks and LaserPointers automatically generate hover and mouse events on Overlays
- turn off the controllerScripts to show that the new C++ generate events work better than the old events from JS

To demo:
- Run default scripts from this PR and [this test script](https://gist.githubusercontent.com/SamGondelman/733b70ce3cac30a0dc2d3494442f2985/raw/fdaf6b00519672aaa6d9486c156e41a6d9ecb54a/UITest.js).  The script creates a big Web3DOverlay in front of you with the same content as the tablet.  In HMD, using your right hand, UI interaction on that overlay with the laser should be very responsive, regardless of game rate.

(I've found that one reason the tablet interaction is so messed up right now is that there's a bug in master where the overlayLaserInput module doesn't properly ignore the tablet model when doing ray picks, so there's z-fighting between the screen and the back.  You can see this if you bring up the tablet in this PR; interaction with it is much buggier than with the big overlay the test script makes)

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4020/HighFidelity-Beta-PR11564-162b5e2c412a742e9ca4a66ff244d500e3ab2a4d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4020/HighFidelity-Beta-PR11564-162b5e2c412a742e9ca4a66ff244d500e3ab2a4d.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4032/HighFidelity-Beta-PR11564-19e5fc4806b0af8f234cc4c1cfe8fe4b6b1e703a.dmg

File: interface/src/raypick/RayPickManager.cpp [hunk 1]
```diff
@@ -42,85 +42,222 @@ void RayPickManager::cacheResult(const bool intersects, const RayPickResult& res
...
+}
+
+void sendMousePressEvent(const RayPickResult& res) {
+    uint32_t id = 0;
+    glm::vec3 direction = -res.surfaceNormal;
+    PointerEvent pointerEvent = PointerEvent(PointerEvent::Press, id,
+        projectOntoOverlayXYPlane(res.objectID, res.intersection),
+        res.intersection, res.surfaceNormal, direction, PointerEvent::PrimaryButton);
+    //qDebug() << "sendMousePressEvent" << projectOntoOverlayXYPlane(res.objectID, res.intersection);
+    emit qApp->getOverlays().mousePressOnOverlay(res.objectID, pointerEvent);
```
Comment (SamGondelman) [inline interface/src/raypick/RayPickManager.cpp]: This is one example of how silly the PointerEvent path is right now.  In scripts, we're currently doing:
- `Overlays.sendMousePressOnOverlay(overlayID, pointerEvent)` -> emit `mousePressOnOverlay` -> every Web3DOverlay connects to the signal and checks if its ID matches overlayID -> send mouse event to web surface

Here, I've cut out the first step, but we should really just be looking up the overlay by ID and directly calling `mousePressOnOverlay` on it.

File: interface/src/ui/overlays/Web3DOverlay.cpp [hunk 2]
```diff
@@ -161,35 +161,29 @@ void Web3DOverlay::buildWebSurface() {
...
-        auto self = weakSelf.lock();
-        if (!self) {
-            return;
-        }
-        if (overlayID == selfOverlayID && (self->_pressed || (!self->_activeTouchPoints.empty() && self->_touchBeginAccepted))) {
-            PointerEvent endEvent(PointerEvent::Release, event.getID(), event.getPos2D(), event.getPos3D(), event.getNormal(), event.getDirection(),
-                event.getButton(), event.getButtons(), event.getKeyboardModifiers());
-            forwardPointerEvent(overlayID, endEvent);
-        }
-    });
```
Comment (SamGondelman) [inline interface/src/ui/overlays/Web3DOverlay.cpp]: I can't see any reason for these slots to be like this.

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4034/HighFidelity-Beta-PR11564-19e5fc4806b0af8f234cc4c1cfe8fe4b6b1e703a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4034/HighFidelity-Beta-PR11564-19e5fc4806b0af8f234cc4c1cfe8fe4b6b1e703a.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4047/HighFidelity-Beta-PR11564-bef25b7028058861d8adb56cf8ef5e078682335e.dmg

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4050/HighFidelity-Beta-PR11564-bef25b7028058861d8adb56cf8ef5e078682335e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4050/HighFidelity-Beta-PR11564-bef25b7028058861d8adb56cf8ef5e078682335e.exe

**Pull request closed without merging**